### ISSUE TITLE
Ubuntu 16.04 LTS Support

### ISSUE THREAD

**Issue by MylesGray** [pi-hole/pi-hole#519]
##### Behaviour:
Installation doesn't succeed on Ubuntu 16.04 LTS due to dependency changes.

##### Steps to reproduce this issue:
Fresh Ubuntu 16.04 LTS install
Run command - fails to start lighttpd due to php-cgi not present.

Can be worked around by manually installing missing packages.

**Comment by dschaper** [comment]
Thanks for the report. Do you have a list of the packages that are missing from the standard LTS install? Is it just the PHP subsystem that is missing?

**Comment by MylesGray** [comment]
PR Submitted.

**Comment by diginc** [comment]
@MylesGray my docker image should run on 16.04 if you install docker first.  https://hub.docker.com/r/diginc/pi-hole/

**Comment by ianrobrien** [comment]
I had the same issue on Ubuntu Mate and it was with php. I solved this by running:
`sudo apt-get install php-cgi php apache2`

**Comment by dschaper** [comment]
This was resolved with the PHP5 vs PHP7 solution. Closing this out for now. Thanks for all the submissions!

**Comment by jpluimers** [comment]
It might not be solved after all. This is a translated from Dutch message at https://tweakers.net/downloads/38419/pi-hole-29.html?showReaction=8982193#r_8982193 talking about the 2.9 version:

> For Ubuntu 16.04 users: you have to install apache and php yourself as it doesn't provide `php5`, so: `sudo apt install php-cgi php apache2`
> 
> After that you can (re)install Pi-hole without trouble.
> 
> Edit:
> An even better way is to modify the downloaded [installscript](https://raw.githubusercontent.com/pi-hole/pi-hole/master/automated%20install/basic-install.sh) and change this line:
> `PIHOLE_DEPS=( dnsutils bc dnsmasq lighttpd php5-common php5-cgi php5 git curl unzip wget sudo netcat )`
> into:
> `PIHOLE_DEPS=( dnsutils bc dnsmasq lighttpd php-common php-cgi php git curl unzip wget sudo netcat )`
> So you change `php5` into `php`
> 
> after that you can run the installer:
> `cat basic-install.sh | bash`

(I don't run ubuntu, I'm just the messenger)

**Comment by dschaper** [comment]
Well, that's installing Apache, which isn't the default install, we use `lighttpd`. So it looks like that instruction is installing Apache2 and PHP7, instead of PHP5 and lighttpd. Which is fine if that's the webserver you'd like to use. There is no problem with PHP7, it's just with an earlier version of PHP5 that is missing a hash comparison function.

**Comment by oregonisaac** [comment]
Same issue when using Ubuntu 16.04.3:

[✗] Update local cache of available packages
Error: Unable to update package cache. Please try "\e[1;31m"

**Comment by dschaper** [comment]
Run `sudo apt update` and see what error is generated by `apt`.

**Comment by oregonisaac** [comment]
No errors generated.  I have pi-hole up and running though, great software.
I just commented out that section of the script and manually ran sudo apt update prior to running the bash script.
FYI the script was also not able to "update" the network configuration file as the location in the script does not exist.  I manually created that file in order to get the second part of the script (after network configuration) to run.
